src/main/java/codeu/model/data/User.java

```diff
@@ -144,8 +144,7 @@ public void setPals(List<String> pals) {
     /** Adds a pal to the list of this User's pals as long as the pal is not this User. */
     public void addPal(String name) {
         /* Validate user input & only add the pal if he/she isn't this User and is not already a pal. */
-        UserStore userStore = UserStore.getInstance();
-        if (name.matches("[\\w*\\s*]*") && userStore.isUserRegistered(name) && !name.equals(this.name) && !isPal(name)) {
+        if (name.matches("[\\w*\\s*]*") && !name.equals(this.name) && !isPal(name)) {
             this.pals.add(name);
```

Comment:
You realise List<String> may not be the best choice for pals. So far at least you aren't using the functionality of a List, and when you do pals.contains(..) in the isPal() method it has to iterate the entire list, hence O(n) efficiency. If add() and contains() are the only operations you need, a Set (eg HashSet) might be a better choice.